File: Algorithm/Algorithm/Random.cs
using System;
using System.Collections.Generic;
using Pair = System.Tuple<int, int>;
using PairList = System.Collections.Generic.List<System.Tuple<int, int>>;
using Directions = System.Collections.Generic.List<System.Collections.Generic.List<System.Collections.Generic.List<char>>>;
using Bools = System.Collections.Generic.List<System.Collections.Generic.List<bool>>;

namespace Algorithm
{
    public class Random
    {
        private long seed;
        public long InitialSeed { get; }

        public Random(int seed)
        {
            this.seed = seed;
        }

        public int Generate(long low, long high)
        {
            long mod = high - low + 1;
            if(seed < 1)
            {
                seed = (low + high - seed + 1999) * 1582307 % 55555333;
            }
            seed = (3515366 * seed + 12345) % 99999989;
            int value = (int)((seed % mod) + low);
            return value;
        }

        public int ChooseFrom(PairList ranges)
        {
            int choice = Generate(ranges[0].Item1, ranges[ranges.Count-1].Item2);
            for(int i = 0; i < ranges.Count; i++)
            {
                if(Util.Between(choice, ranges[i]))
                {
                    return i;
                }
            }
            return -1;
        }
    }
}

File: python/random.py
import util

class RandomSeed:

    def __init__(self, seed):
        self.seed = seed
        self.initial_seed = seed

    def generate(self, low, high):
        mod = high - low + 1
        if self.seed < 1: # Either 0 or negative, out of bounds due to a bad initial seed probably
            self.seed =  ( (low + high - self.seed + 1999) * 1582307 ) % 55555333
        self.seed = (3515366 * self.seed + 12345) % 99999989
        # print self.seed
        value = (self.seed % mod) + low
        return value

    def choose_from(self, tuple_ranges):
        choice = self.generate(tuple_ranges[0][0], tuple_ranges[-1][1])
        for i in range(len(tuple_ranges)):
            if util.between(choice, tuple_ranges[i]):
                return i
        return None
